test/Cashier.test.js

```diff
@@ -100,4 +143,382 @@ describe('Cashier', function () {
+      // ERC20
+      const someoneNativeTokenAfter = await web3.eth.getBalance(
+        someone
+      )
+      expect(someoneNativeTokenAfter).to.not.equal(someoneNativeTokenBefore);
+
+      const contractNativeTokenAfter = await web3.eth.getBalance(
+        this.contract.address
+      )            
+      expect(contractNativeTokenAfter).to.not.equal(contractNativeTokenBefore);
+
+      
+      // Reserve
+      await anotherOneReserveTracker.requireIncrease(toBN(amountReserve));      
+      await contractReserveTracker.requireDecrease(toBN(amountReserve)); 
+      
+      /// check event
+      expectEvent(tx, 'Deposit', { 
+        'token': ZERO_ADDRESS,
+        'from': someone,
+        'to': anotherOne,
+        'srcAmount': amountSrc,
+        'reserveAmount': amountReserve,
+      });     
+
+    });
+
+    it('Should NOT collect balances if caller is not the owner', async function(){
+      
+      tokens = [this.ERC20token.address]
+      collectorAdd = owner
+      
+      await expectRevert( 
+        this.contract.collectBalances(
+          tokens,
+          collectorAdd,
+          {
+            from: someone,
+          }
+      ), 'Ownable: caller is not the owner')
+
+    });
+
+    it('Should OK collect ERC20 balances', async function(){
+      tokens = [this.ERC20token.address]
+      collectorAdd = owner
+      amountSrc = toWei('10') //ERC20 amount already in contract
+
+      const collectorERC20Tracker = await balanceSnap(
+        this.ERC20token, collectorAdd, 'collector\s reserve'
+      );       
+      const contractERC20Tracker = await balanceSnap(
+        this.ERC20token, this.contract.address, 'contract\s reserve'
+      );               
+      
+      await this.contract.collectBalances(
+          tokens,
+          collectorAdd,
+          {
+            from: owner,
+          }
+      );
+
+      // increase - decrease asserts
+      // ERC20
+      await contractERC20Tracker.requireDecrease(toBN(amountSrc));      
+      await collectorERC20Tracker.requireIncrease(toBN(amountSrc));      
+    });
+
+    it('Should OK collect Native token balance', async function(){
+      tokens = [ZERO_ADDRESS]
+      collectorAdd = owner
+      amountSrc = toWei('10') //Native token amount already in contract
+
+      const ownerNativeTokenBefore = await web3.eth.getBalance(
+        owner
+      )
+      const contractNativeTokenBefore = await web3.eth.getBalance(
+        this.contract.address
+      )
+
+      tx = await this.contract.collectBalances(
+        tokens,
+        collectorAdd,
+        {
+          from: owner,
+        }
+      );      
+
+      const ownerNativeTokenAfter = await web3.eth.getBalance(
+        owner
+      )
+      const contractNativeTokenAfter = await web3.eth.getBalance(
+        this.contract.address
+      )     
+
+      expect(ownerNativeTokenAfter).to.not.equal(ownerNativeTokenBefore);
+      expect(contractNativeTokenAfter).to.not.equal(contractNativeTokenBefore);
+
```

Comment:
This native token asserts should be changed @chentschel . 
We need more precision